src/main/java/org/folio/rs/service/ReturnItemService.java

```diff
@@ -0,0 +1,98 @@
+package org.folio.rs.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.rs.client.CirculationClient;
+import org.folio.rs.client.InventoryClient;
+import org.folio.rs.client.UsersClient;
+import org.folio.rs.domain.dto.CheckInItem;
+import org.folio.rs.domain.dto.ContributorName;
+import org.folio.rs.domain.dto.EffectiveCallNumberComponents;
+import org.folio.rs.domain.dto.Item;
+import org.folio.rs.domain.dto.Request;
+import org.folio.rs.domain.dto.ReturnItemResponse;
+import org.folio.rs.domain.dto.User;
+import org.folio.rs.domain.entity.RetrievalQueueRecord;
+import org.folio.rs.error.ItemReturnException;
+import org.folio.rs.repository.RetrievalQueueRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toList;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class ReturnItemService {
+
+  private static final String BARCODE_QUERY_PROPERTY = "barcode==";
+  private static final String USER_ID_QUERY_PROPERTY = "id==";
+
+  private final InventoryClient inventoryClient;
+  private final CirculationClient circulationClient;
+  private final UsersClient usersClient;
+  private final RetrievalQueueRepository retrievalQueueRepository;
+  private final CheckInItemService checkInItemService;
+
+  public ReturnItemResponse returnItem(String remoteStorageConfigurationId, CheckInItem checkInItem) {
```

Comment:
Could you please refactor this method, for example, introducing separate methods for logic parts. Current method complexity make difficult to go through business logic.